### PR TITLE
Add waiting time to gate evaluate command

### PR DESCRIPTION
**Pull request by andrea-mosk**
### What and why?

Add waiting time to gate evaluate command

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/gate/README.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ datadog-ci gate evaluate --scope team:backend --scope team:frontend --fail-on-em
 - `--dry-run` runs the command without the final evaluation step. All other checks are performed. The default value is `false`.
 - `--fail-if-unavailable` fails the command if Datadog is unavailable. The default value is `false`.
 - `--fail-on-empty` fails the command if no matching rules are found in Datadog. The default value is `false`.
+- `--no-wait` can be used in order to remove the default waiting time (30s) that the command has to ensure that the events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of the events. Read more on the quality gates documentation before enabling this. The default value is `false`.
```
Comment (juan-fernandez) [inline src/commands/gate/README.md:28]: I think this is a bit clearer:
```suggestion
- `--no-wait` removes the waiting time (30s) that is in place to ensure that events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.
```

Also, why do we want to allow users to set this? Isn't it dangerous?

**Review by liashenko — APPROVED** [review]

File: src/commands/gate/README.md (hunk 1 above)
Comment (andrea-mosk) [reply src/commands/gate/README.md]: Looks clearer, thanks! 

> Also, why do we want to allow users to set this? Isn't it dangerous?

It's not really dangerous, what can happen is that rules might be incorrectly evaluated, but we need to do a good job on the documentation side on making sure they understand the risks. There are many use cases in which they wouldn't want to wait for 30s, for example when there are intermediate jobs between when the events are sent and when `datadog-ci gate evaluate` is called.

**Review by juan-fernandez — APPROVED** [review]

File: src/commands/gate/README.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ datadog-ci gate evaluate --scope team:backend --scope team:frontend --fail-on-em
 - `--dry-run` runs the command without the final evaluation step. All other checks are performed. The default value is `false`.
 - `--fail-if-unavailable` fails the command if Datadog is unavailable. The default value is `false`.
 - `--fail-on-empty` fails the command if no matching rules are found in Datadog. The default value is `false`.
+- `--no-wait` removes the waiting time (30s) that is in place to ensure that events (e.g. tests) can be properly queried by the rules. This mechanism avoids that rules are incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.
```
Comment (cswatt) [inline src/commands/gate/README.md:28]:
```suggestion
- `--no-wait` removes the waiting time (30s) that is in place to ensure that events (for example: tests) can be properly queried by the rules. This mechanism prevents rules from being incorrectly evaluated due to the absence of events. Read more on the quality gates documentation before passing this flag. The default value is `false`.
```

would be nice to add a link to the quality gates documentation

**Review by cswatt — APPROVED** [review]
leaving feedback on docs, approving so as not to block

File: src/commands/gate/README.md (hunk 2 above)
Comment (andrea-mosk) [reply src/commands/gate/README.md]: Thanks, I have committed your suggestion.  We still need to create the documentation, so no link for now 😬

**Pull request merged**